### DISCUSSION TITLE
zoom and scroll page

### DISCUSSION THREAD

**Discussion by Turtleted21** [google/model-viewer#1929] [Q&A]
Hello
Is it possible to ensure that the modelviewer background don't receive the scroll, only 3D model?
Because I have a scroll page and if my mouse is inside modelviewer div it zoom instead of scrolling on the page.
I want to keep the zoom functionality but aslo scroll to my page.

Or perhaps another solution?

Thanks

![scrollpage](https://user-images.githubusercontent.com/46889785/97563636-a0254600-19e3-11eb-85ae-418a7339e98b.png)

**Comment by elalish** [comment]
Can I assume you're using the latest, v1.3? Perhaps you want to set the [touch-action](https://modelviewer.dev/docs/#entrydocs-stagingandcameras-attributes-touch-action) attribute to `'none'`? I'm not entirely sure what behavior you're after. When you say mouse zoom, at you talking about a mouse wheel? And do you really want both zooming and scrolling to happen at once? You can also set up zoom to be triggered by scroll [directly](https://modelviewer.dev/examples/stagingandcameras/#orbitAndScroll).

**Comment by Turtleted21** [comment]
_Can I assume you're using the latest, v1.3?_
yes

 _Perhaps you want to set the [touch-action](https://modelviewer.dev/docs/#entrydocs-stagingandcameras-attributes-touch-action) attribute to `'none'`?_ 
I don't see any effect, perhaps it's only for mobile

_When you say mouse zoom, at you talking about a mouse wheel?_
yes

 _And do you really want both zooming and scrolling to happen at once? You can also set up zoom to be triggered by scroll_
Not at once, I want my mouse wheel zoom only if my mouse cursor is over my 3d model, but scroll my page if I'm out.
Currently my mouse wheel zoom also if my mouse cursor is over the model-viewer background, but like my model-viewer background is transparent, the user don't really know if he over the model-viewer to zoom or over the page background to scroll.

you can see my test page here : https://data.strategia.io/modelviewertest/index.html

I think it need a line trace hit between 3d model and mouse cursor to detect if i'm over the 3d model or not, I don't know if model-viewer can do that.

**Comment by elalish** [comment]
Ah, thank you for the details, I understand now. We don't support hit-testing to determine what interactions do for a couple of reasons. First, it's expensive to hit test, so putting that in a mouse handler would destroy page responsiveness. Second, some models (like fences or screens) have such finely detailed geometry that a user could have a hard time putting their mouse on/off the model reliably. Perhaps a subtle background highlight when the user is hovering over the model-viewer element?

**Comment by ZeusLT** [comment]
What would I have to do to disable model-viewer zoom with scroll wheel and let it scroll the page instead?

**Comment by elalish** [comment]
At the moment all we have is not including the `camera-controls` attribute, but then you lose rotation as well.
